packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java

```diff
@@ -156,13 +158,17 @@ private Element doRender(VariableManager nodeVariableManager, String targetObjec
         String targetObjectLabel = nodeDescription.getTargetObjectLabelProvider().apply(nodeVariableManager);
         Set<ViewModifier> modifiers = this.computeModifiers(optionalDiagramEvent, optionalPreviousNode, nodeId);
         ViewModifier state = this.computeState(modifiers);
+        CollapsingState collapsingState = this.shouldRenderChildren(nodeId, optionalPreviousNode, optionalDiagramEvent);
```

Comment:
The style provider does not seems to have access to the `collapsingState`, I'll update that and provide an example in Papaya